File: Explorer/MicrosoftOrganizationChartReader/Records/GlobalInfoRecord.cs
using System.ComponentModel;
using System.Drawing;
using System.IO;

namespace OrganizationChartReader
{
    public class GlobalInfoRecord
    {
        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown1 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown2 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown3 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown4 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown5 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown6 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown7 { get; set; }

        public GlobalGroupsStyle GroupsStyle { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown8 { get; set; }

        public GlobalInfoRecord(BinaryReader reader)
        {
            Unknown1 = reader.ReadUInt16();
            Unknown2 = reader.ReadUInt16();
            Unknown3 = reader.ReadUInt16();
            Unknown4 = reader.ReadUInt16();
            Unknown5 = reader.ReadUInt16();
            Unknown6 = reader.ReadUInt16();
            Unknown7 = reader.ReadUInt16();
            GroupsStyle = (GlobalGroupsStyle)reader.ReadUInt16();
            Unknown8 = reader.ReadUInt16();
        }

        public enum GlobalGroupsStyle: ushort
        {
            Style1 = 0x00,
            Style2 = 0x02,
            Style3 = 0x03,
            Style4 = 0x04,
            Style5 = 0x05,
            CoManager = 0x22,
        }
    }
}

File: Sources/MicrosoftOrganizationChartReader/Records/GlobalInfoRecord.swift
//
//  GlobalInfoRecord.swift
//
//
//  Created by Hugh Bellamy on 20/01/2021.
//

import DataStream

public struct GlobalInfoRecord {
    public let unknown1: UInt16
    public let unknown2: UInt16
    public let unknown3: UInt16
    public let unknown4: UInt16
    public let unknown5: UInt16
    public let unknown6: UInt16
    public let unknown7: UInt16
    public let groupsStyle: GroupsStyle
    public let unknown8: UInt16
    
    public init(dataStream: inout DataStream) throws {
        self.unknown1 = try dataStream.read(endianess: .littleEndian)
        self.unknown2 = try dataStream.read(endianess: .littleEndian)
        self.unknown3 = try dataStream.read(endianess: .littleEndian)
        self.unknown4 = try dataStream.read(endianess: .littleEndian)
        self.unknown5 = try dataStream.read(endianess: .littleEndian)
        self.unknown6 = try dataStream.read(endianess: .littleEndian)
        self.unknown7 = try dataStream.read(endianess: .littleEndian)
        self.groupsStyle = try GroupsStyle(dataStream: &dataStream)
        self.unknown8 = try dataStream.read(endianess: .littleEndian)
    }
    
    public enum GroupsStyle: UInt16, DataStreamCreatable {
        case style1 = 0x00
        case style2 = 0x02
        case style3 = 0x03
        case style4 = 0x04
        case style5 = 0x05
        case coManager = 0x22
    }
}
